diff --git a/pbwtMatch.c b/pbwtMatch.c
index de28934..ad0ee91 100644
--- a/pbwtMatch.c
+++ b/pbwtMatch.c
@@ -119,25 +119,10 @@ void matchMaximalWithin (PBWT *p, void (*report)(int ai, int bi, int start, int
   PbwtCursor *u = pbwtCursorCreate (p, TRUE, TRUE) ;
 
   for (k = 0 ; k <= p->N ; ++k)
-    { for (i = 0 ; i < u->M ; ++i)
-	{ m = i-1 ; n = i+1 ;
-	  if (u->d[i] <= u->d[i+1])
-	    while (u->d[m+1] <= u->d[i])
-	      if (u->y[m--] == u->y[i] && k < p->N) goto nexti ;
-	  if (u->d[i] >= u->d[i+1])
-	    while (u->d[n] <= u->d[i+1])
-	      if (u->y[n++] == u->y[i] && k < p->N) goto nexti ;
-	  if (matchLengthHist)
-	    ++array(matchLengthHist, (u->d[i]<u->d[i+1]) ? k-u->d[i] : k-u->d[i+1], int) ;
-	  else
-	    { for (j = m+1 ; j < i ; ++j) (*report) (u->a[i], u->a[j], u->d[i], k) ;
-	      for (j = i+1 ; j < n ; ++j) (*report) (u->a[i], u->a[j], u->d[i+1], k) ;
-	    }
-	nexti: ;
-	}
+    {
       pbwtCursorForwardsReadAD (u, k) ;
+      printf("%d\n",u->a[p->M-1] + u->d[p->M-1]);
     }
-
   pbwtCursorDestroy (u) ;
 }
 
